openedx_events/data.py

```diff
@@ -62,7 +75,8 @@ class EventsMetadata:
     source = attr.ib(
         type=str, default=None,
         converter=attr.converters.default_if_none(
-            attr.Factory(lambda: "openedx/{service}/web".format(service=getattr(settings, "SERVICE_VARIANT", "")))
+            attr.Factory(lambda: "openedx/{service}/web".format(service=get_source_name()) if get_source_name()
```

Comment:
NIt: 
1. I'd find this easier to read if on and earlier line you set:
```
service_name = "unidentified" or _get_service_name().  # see other PR comment for method name change
```
2. Consider something like `SERVICE-NAME-UNSET` in place of `unidentified`, to give something easy to search for and easy to resolve.